packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.tsx

```diff
@@ -421,7 +419,7 @@ export class DetailsList extends React.Component<IDetailsListProps, IDetailsList
       if (this.refs.selectionZone) {
         this.refs.selectionZone.ignoreNextFocus();
       }
-      row.focus();
+      this._async.setTimeout(() => row.focus(), 0);
```

Comment:
measurement shows focus cost reduced from 30ms to 3 ms after changing it to async. 
This is help reduce our rendering time to glass.